### PR TITLE
feat(rum): add sumo logic rum 

### PR DESCRIPTION
**Pull request by mat-rumian**
# Changes

* Modify `frontend` app with possibility to load Sumo Logic RUM
* Update `docker-compose` deployment
* Update `helm` deployment

### CODE REVIEW

File: sumo_deployments/docker-compose/otel-config.yaml [hunk 1]
```diff
@@ -16,18 +16,28 @@ receivers:
 
 exporters:
   sumologic:
+    sending_queue:
+      enabled: true
+      num_consumers: 100
```
Comment (swiatekm-sumo) [inline sumo_deployments/docker-compose/otel-config.yaml:21]: Does it need to be this many? We use 10 by default in the Helm Chart.

File: sumo_deployments/helm/sumo-opentelemetry-demo/Chart.yaml [hunk 2]
```diff
@@ -2,12 +2,12 @@ apiVersion: v2
 name: sumo-opentelemetry-demo
 description: A Helm chart for Kubernetes
 type: application
-version: 0.1.0
-appVersion: 0.1.0
+version: 0.1.1
+appVersion: 0.1.1
```
Comment (swiatekm-sumo) [inline sumo_deployments/helm/sumo-opentelemetry-demo/Chart.yaml:6]: Can this be done in a separate change? Do we even have any automation for publishing this Chart?

File: sumo_deployments/helm/sumo-opentelemetry-demo/templates/frontend.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            - name: OTEL_EXPORTER_OTLP_ENDPOINT
+              value: http://{{ include "sumo-opentelemetry-demo-otelagent-svc-name" . }}:4317
+            - name: FRONTEND_PORT
+              value: "8080"
+            - name: SUMO_LOGIC_RUM_ENABLED
+              value: "true"
+            - name: PUBLIC_SUMO_LOGIC_RUM_SERVICE_NAME
+              value: frontend-rum
+            - name: PUBLIC_SUMO_LOGIC_RUM_APPLICATION_NAME
+              value: opentelemetry-demo
```
Comment (swiatekm-sumo) [inline sumo_deployments/helm/sumo-opentelemetry-demo/templates/frontend.yaml:79]: Would be nicer if these were all includes from _helpers.tpl, especially the Service names.

File: sumo_deployments/helm/sumo-opentelemetry-demo/values.yaml [hunk 4]
```diff
@@ -130,6 +133,21 @@ opentelemetry-demo:
         affinity: {}
         tolerations: []
 
+    accountingService:
+      enabled: true
+      useDefault:
+        env: true
+      env:
+        - name: OTEL_EXPORTER_OTLP_ENDPOINT
+          value: 'http://{{ include "sumo-opentelemetry-demo-otelagent-svc-name" . }}:4317'
```
Comment (swiatekm-sumo) [inline sumo_deployments/helm/sumo-opentelemetry-demo/values.yaml:142]: Does this even work? Do you manually handle it somewhere?

**Review by swiatekm-sumo — COMMENTED** [review]
I've mostly reviewed the infrastructure parts, don't feel competent when it comes to the frontend.

File: sumo_deployments/helm/sumo-opentelemetry-demo/Chart.yaml (hunk 2 above)
Comment (mat-rumian) [reply sumo_deployments/helm/sumo-opentelemetry-demo/Chart.yaml:6]: No, chart is not published. We just have it here locally in the sources.

File: sumo_deployments/helm/sumo-opentelemetry-demo/values.yaml (hunk 4 above)
Comment (mat-rumian) [reply sumo_deployments/helm/sumo-opentelemetry-demo/values.yaml:142]: Yes, it works fine and it's implemented [here](https://github.com/SumoLogic/opentelemetry-demo/blob/add-sumo-rum-frontend/sumo_deployments/helm/sumo-opentelemetry-demo/templates/_helpers.tpl#L65)

File: sumo_deployments/helm/sumo-opentelemetry-demo/templates/frontend.yaml (hunk 3 above)
Comment (mat-rumian) [reply sumo_deployments/helm/sumo-opentelemetry-demo/templates/frontend.yaml:79]: I agree but for now the goal is to just run is asap and provide demo data to Sumo to present our capabilities :) I will prepare a TODO task for that in future.

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**